### ISSUE TITLE
include={field} not working on specific object request

### ISSUE THREAD

**Issue by nkranes** [ParsePlatform/parse-server#1124]
For implementation related questions or technical support, please refer to the [Stack Overflow](http://stackoverflow.com/questions/tagged/parse.com) and [Server Fault](https://serverfault.com/tags/parse) communities.

Make sure these boxes are checked before submitting your issue -- thanks for reporting issues back to Parse Server!

- [x] You've met the [prerequisites](https://github.com/ParsePlatform/parse-server/wiki/Parse-Server-Guide#prerequisites).

- [X] You're running the [latest version](https://github.com/ParsePlatform/parse-server/releases) of Parse Server.

- [X] You've searched through [existing issues](https://github.com/ParsePlatform/parse-server/issues?utf8=%E2%9C%93&q=). Chances are that your issue has been reported or resolved before.

#### Environment Setup
parse-server running locally

#### Steps to reproduce
1 - Insert a class
curl -X POST -H "X-Parse-Application-Id: XXXX" -H "Content-Type: application/json" -H "X-Parse-Master-Key: XXXX" -d '{
    "platform" : "Android",
    "isMerged" : "true"
}' "http://localhost:1337/parse/classes/AppConfig"

ObjectId: uq9RVGSIrv

2 - Insert another class
curl -X POST -H "X-Parse-Application-Id: XXXX" -H "Content-Type: application/json" -H "X-Parse-Master-Key: XXXX" -d '{
    "name" : "John Smith",
    "streetAddress" : "123 Main St"
}' "http://localhost:1337/parse/classes/Player"

ObjectId: UoBKuO29mp

3 - Insert a few types of relations between the 2 classes, since I'm not sure what the correct syntax is
curl -X PUT -H "X-Parse-Application-Id: XXXX" -H "Content-Type: application/json" -H "X-Parse-Master-Key: XXXX" -d '{"player":{"__op":"AddUnique","objects":[{"__type":"Pointer","className":"Player","objectId":"UoBKuO29mp"}]}}
' "http://localhost:1337/parse/classes/AppConfig/uq9RVGSIrv"

curl -X PUT -H "X-Parse-Application-Id: XXXX" -H "Content-Type: application/json" -H "X-Parse-Master-Key: XXXX" -d '{"player2":{"__op":"AddRelation","objects":[{"__type":"Pointer","className":"Player","objectId":"UoBKuO29mp"}]}}
' "http://localhost:1337/parse/classes/AppConfig/uq9RVGSIrv"

curl -X PUT -H "X-Parse-Application-Id: XXXX" -H "Content-Type: application/json" -d '{"player3":{"__op":"Add","objects":[{"__type":"Pointer","className":"Player","objectId":"UoBKuO29mp"}]}}
' "http://localhost:1337/parse/classes/AppConfig/uq9RVGSIrv"

curl -X POST -H "X-Parse-Application-Id: XXXX" -H "Content-Type: application/json" -d '{"Level" : "TWO", "player4":{"__op":"AddUnique","objects":[{"__type":"Pointer","className":"Player","objectId":"UoBKuO29mp"}]}}
' "http://localhost:1337/parse/classes/AppConfig"

4 - Get the AppConfig object and notice the "include=" does not work
curl -X GET -H "X-Parse-Application-Id: XXXX" -H "Content-Type: application/json" "http://localhost:1337/parse/classes/AppConfig/uq9RVGSIrv?include=player"

Results:
```
{
  "objectId": "uq9RVGSIrv",
  "platform": "Android",
  "isMerged": "true",
  "updatedAt": "2016-03-21T17:47:11.682Z",
  "createdAt": "2016-03-21T16:56:14.816Z",
  "player": [
    {
      "__type": "Pointer",
      "className": "Player",
      "objectId": "UoBKuO29mp"
    }
  ],
  "player3": [
    {
      "__type": "Pointer",
      "className": "Player",
      "objectId": "UoBKuO29mp"
    }
  ]
}

```
5 - Get rid of the objectId in the query and it works as expected
curl -X GET -H "X-Parse-Application-Id: XXXX" -H "Content-Type: application/json" "http://localhost:1337/parse/classes/AppConfig?include=player"

```
{
  "results": [
    {
      "objectId": "uq9RVGSIrv",
      "platform": "Android",
      "isMerged": "true",
      "updatedAt": "2016-03-21T17:47:11.682Z",
      "createdAt": "2016-03-21T16:56:14.816Z",
      "player": [
        {
          "objectId": "UoBKuO29mp",
          "name": "John Smith",
          "streetAddress": "123 Main St",
          "updatedAt": "2016-03-21T16:54:44.271Z",
          "createdAt": "2016-03-21T16:54:44.271Z",
          "__type": "Object",
          "className": "Player"
        }
      ],
      "player3": [
        {
          "__type": "Pointer",
          "className": "Player",
          "objectId": "UoBKuO29mp"
        }
      ]
    },
    {
      "objectId": "UvBVu5EKB3",
      "Level": "TWO",
      "player4": [
        {
          "__type": "Pointer",
          "className": "Player",
          "objectId": "UoBKuO29mp"
        }
      ],
      "updatedAt": "2016-03-21T17:49:20.945Z",
      "createdAt": "2016-03-21T17:49:20.945Z"
    }
  ]
}

```
OR

curl -X GET -H "X-Parse-Application-Id: XXXX" -H "Content-Type: application/json" "http://localhost:1337/parse/classes/AppConfig?include=player&include=player4&include=player3"

```
{
  "results": [
    {
      "objectId": "uq9RVGSIrv",
      "platform": "Android",
      "isMerged": "true",
      "updatedAt": "2016-03-21T17:47:11.682Z",
      "createdAt": "2016-03-21T16:56:14.816Z",
      "player": [
        {
          "objectId": "UoBKuO29mp",
          "name": "John Smith",
          "streetAddress": "123 Main St",
          "updatedAt": "2016-03-21T16:54:44.271Z",
          "createdAt": "2016-03-21T16:54:44.271Z",
          "__type": "Object",
          "className": "Player"
        }
      ],
      "player3": [
        {
          "objectId": "UoBKuO29mp",
          "name": "John Smith",
          "streetAddress": "123 Main St",
          "updatedAt": "2016-03-21T16:54:44.271Z",
          "createdAt": "2016-03-21T16:54:44.271Z",
          "__type": "Object",
          "className": "Player"
        }
      ]
    },
    {
      "objectId": "UvBVu5EKB3",
      "Level": "TWO",
      "player4": [
        {
          "objectId": "UoBKuO29mp",
          "name": "John Smith",
          "streetAddress": "123 Main St",
          "updatedAt": "2016-03-21T16:54:44.271Z",
          "createdAt": "2016-03-21T16:54:44.271Z",
          "__type": "Object",
          "className": "Player"
        }
      ],
      "updatedAt": "2016-03-21T17:49:20.945Z",
      "createdAt": "2016-03-21T17:49:20.945Z"
    }
  ]
}

```
I think this is not the expected behavior. 

Thanks!

**Comment by oleksiy-nesterov** [comment]
confirm

**Comment by dtsymbal** [comment]
2.2.1 - bug exists

**Comment by jeremyjackson89** [comment]
This is an issue for me as well -- 2.2.0

**Comment by andrew-makarenko** [comment]
confirm

**Comment by kos984** [comment]
confirm

**Comment by jeremyjackson89** [comment]
## Temp Fix
This is working for me at the moment, at least until the parse team provides an update. Hopefully this helps some of you as well.

If you go into the code and look inside of `parse-server/lib/Routers/ClassesRouter.js` there's an array with objects that have key value pairs for functions. On 2.2.2 on line 137 there's this bit of code:

```
key: 'handleGet',
value: function handleGet(req) {
    return _rest2.default.find(req.config, req.auth, req.params.className, { objectId: req.params.objectId }).then(function(response) {
        if (!response.results || response.results.length == 0) {
            throw new Parse.Error(Parse.Error.OBJECT_NOT_FOUND, 'Object not found.');
        }

        if (req.params.className === "_User") {

            delete response.results[0].sessionToken;

            var user = response.results[0];

            if (req.auth.user && user.objectId == req.auth.user.id) {
                // Force the session token
                response.results[0].sessionToken = req.info.sessionToken;
            }
        }
        return {
            response: response.results[0]
        };
    });
}
```

If you look at the method above it on line 42 (`handleFind`), you can see where they're parsing through the query parameters. So I took some of that and added it to `handleGet()`:

```
key: 'handleGet',
value: function handleGet(req) {
  var body = Object.assign(req.body, ClassesRouter.JSONFromQuery(req.query));
  var options = {};
  var allowConstraints = ['include'];

  var _iteratorNormalCompletion = true;
  var _didIteratorError = false;
  var _iteratorError = undefined;

  try {
    for (var _iterator = Object.keys(body)[Symbol.iterator](), _step; !(_iteratorNormalCompletion = (_step = _iterator.next()).done); _iteratorNormalCompletion = true) {
      var key = _step.value;

      if (allowConstraints.indexOf(key) === -1) {
        throw new Parse.Error(Parse.Error.INVALID_QUERY, 'Improper encode of parameter');
      }
    }
  } catch (err) {
    _didIteratorError = true;
    _iteratorError = err;
  } finally {
    try {
      if (!_iteratorNormalCompletion && _iterator.return) {
        _iterator.return();
      }
    } finally {
      if (_didIteratorError) {
        throw _iteratorError;
      }
    }
  }

  if (body.include) {
    options.include = String(body.include);
  }

  return _rest2.default.find(req.config, req.auth, req.params.className, { objectId: req.params.objectId }, options).then(function (response) {
    if (!response.results || response.results.length == 0) {
      throw new Parse.Error(Parse.Error.OBJECT_NOT_FOUND, 'Object not found.');
    }

    if (req.params.className === "_User") {

      delete response.results[0].sessionToken;

      var user = response.results[0];

      if (req.auth.user && user.objectId == req.auth.user.id) {
        // Force the session token
        response.results[0].sessionToken = req.info.sessionToken;
      }
    }
    return { response: response.results[0] };
  });
}
```

**Comment by hramos** [comment]
No need to wait for the Parse team, you can open a pull request :) Make sure to also include tests in your PR.

**Comment by flovilmart** [comment]
I believe this has been merged with #1280.

Please reopen if it wasn't
